### DISCUSSION TITLE
How to set a field to be null without being forced to use undefined?

### DISCUSSION THREAD

**Discussion by arleneadriannego** [prisma/prisma#2587] [Q&A] [answered]
Hi! I set a field in a model to be nullable, since it is optional. Before I updated to prisma beta 6, I can save without fuss. Now, after I upgraded to beta 6, I get `Argument role for create.role must not be null. Please use undefined instead.` How do I set this field to `null` without making it `undefined`?

```
model users {
  id          Int     @id @default(autoincrement())
  first_name  String
  role        String?
}
```

**Comment by ryands17** [comment, marked as answer]
Hey @adrn03 :wave: 
I have run the above model on `beta.6` and I'm not facing any issue.

Could you try the below points and let me know if it works for you:
1. Check if both `@prisma/cli` and `@prisma/client` are on `beta.6`.
2. Remove the migrations folder altogether along with the `_Migrations` table in the DB and migrate again.

**Reply by arleneadriannego** [reply]
Thank you! deleting the _Migrations worked

**Reply by rujorgensen** [reply]
Deleting migrations worked for me too, although I'm somewhat concerned about what to do if this happens on my production environment 🤔. Does anybody have a clue as to what causes this?

**Comment by AlexMeah** [comment]
I'm getting this error with "@prisma/client": "2.11.0". It's impossible to set a nullable field to null, `disconnect` throws an error on null field and `undefined` foes nothing.

```ts
{
  where: {
    id: 16
  },
  data: {
    businessInfo: {
      update: {
        logo: null
              ~~~~
      }
    }
  }
}

Argument logo for data.businessInfo.update.logo must not be null. Please use undefined instead.
```

Model: 

```prisma
model BusinessInfo {
  id          Int      @id @default(autoincrement())
  ...
  mediaId     Int?
  logo        Media?   @relation(fields: [mediaId], references: [id])
}
```

**Reply by ryands17** [reply]
As `logo` is a relation and not a value, you just need to pass this:

```ts
{
  where: {
    id: 16
  },
  data: {
    businessInfo: {
      update: {
        logo: { disconnect: true }
      }
    }
  }
}
```

**Comment by ludralph** [comment]
Hi @arleneadriannego,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
